src/firefly/java/edu/caltech/ipac/firefly/server/query/ObsCorePackager.java

```diff
@@ -201,13 +206,11 @@ public static List<FileInfo> parseDatalink(URL url, DownloadRequest request, Map
                         String extension = "unknown";
                         extension = content_type != null ? getExtFromURL(content_type) : getExtFromURL(productUrl);
                         ext_file_name += "." + extension;
-                        ext_file_name = ("folder".equalsIgnoreCase(zipType) || (zipType == null && makeDataLinkFolder))
-                                ? prepend_file_name + "/" + ext_file_name
-                                : ext_file_name;
+                        ext_file_name = ("flat".equalsIgnoreCase(zipType) || (!makeDataLinkFolder)) ?
+                                ext_file_name : prepend_file_name + "/" + ext_file_name;
                     } else {
-                        ext_file_name = ("folder".equalsIgnoreCase(zipType) || (zipType == null && makeDataLinkFolder))
-                                ? (prepend_file_name + "/" + (fileName != null ? fileName : ""))
-                                : fileName;
+                        ext_file_name = ("flat".equalsIgnoreCase(zipType) || (!makeDataLinkFolder)) ?
+                                ext_file_name : (prepend_file_name + "/" + (fileName != null ? fileName : ""));
```

Comment:
same here

Reply:
Yes good point, this is a nice suggestion. The reason I was initially hesitant is because this change will need to be made in other apps' download processors as well (for example `WiseFileGroupsProcessor.java` line 84. But it's the same logic there as well, so I can change this in those apps' code also.

Reply:
Yeah that line in Wise is also converting list of two strings to boolean. The whole idea is to type your params how they are behaving, both options are mutually exclusive